### ISSUE TITLE
No match with standard(?) weather entity

### ISSUE THREAD

**Issue by vingerha** [Makin-Things/platinum-weather-card#7]
using v.0.2
As per your request on feedback from other areas... I am happy to test, live in France and have used many cards which all showed the data as per the (apparently?) standard weather integration. 
With this card almost nothing shows, wrong figures, errors in the card, etc. The selection of entity is a bit odd too, it sometimes pcks up on 'weather' but then switches to zone.

I am pasting my entity below so that you can use this as test-data if you like to analyze as to the 'why' :) 
What I am missing is 'hourly' data too but this uses a different set of data...t.b.d. after this works 

![image](https://user-images.githubusercontent.com/44190435/177084556-8b97dfa7-5db7-45f2-86e2-76a1d2bcffa5.png)

```
temperature: 22.5
humidity: 45
pressure: 1015.1
wind_speed: 4
forecast:
  - datetime: '2022-07-04T02:00:00+02:00'
    condition: sunny
    temperature: 32.4
    templow: 21.3
    precipitation: 0
  - datetime: '2022-07-05T02:00:00+02:00'
    condition: sunny
    temperature: 32
    templow: 20.6
    precipitation: 0
  - datetime: '2022-07-06T02:00:00+02:00'
    condition: sunny
    temperature: 30.2
    templow: 20.7
    precipitation: 0
  - datetime: '2022-07-07T02:00:00+02:00'
    condition: sunny
    temperature: 31.6
    templow: 21.2
    precipitation: 0
  - datetime: '2022-07-08T02:00:00+02:00'
    condition: sunny
    temperature: 30
    templow: 22.1
    precipitation: 0
  - datetime: '2022-07-09T02:00:00+02:00'
    condition: sunny
    temperature: 30.4
    templow: 22.2
    precipitation: 0
  - datetime: '2022-07-10T02:00:00+02:00'
    condition: sunny
    temperature: 30
    templow: 21.9
    precipitation: 0
  - datetime: '2022-07-11T02:00:00+02:00'
    condition: sunny
    temperature: 29.8
    templow: 22.3
    precipitation: 0
  - datetime: '2022-07-12T02:00:00+02:00'
    condition: sunny
    temperature: 30.5
    templow: 22.5
    precipitation: 0
  - datetime: '2022-07-13T02:00:00+02:00'
    condition: sunny
    temperature: 31.1
    templow: 22.5
    precipitation: 0
  - datetime: '2022-07-14T02:00:00+02:00'
    condition: sunny
    temperature: 31.4
    templow: 23
    precipitation: 0
  - datetime: '2022-07-15T02:00:00+02:00'
    condition: sunny
    temperature: 31.5
    templow: 23.2
    precipitation: 0
  - datetime: '2022-07-16T02:00:00+02:00'
    condition: sunny
    temperature: 31.7
    templow: 23.2
    precipitation: 0
  - datetime: '2022-07-17T02:00:00+02:00'
    condition: sunny
    temperature: 31.6
    templow: 23.2
    precipitation: 0
attribution: Data provided by Météo-France
friendly_name: pickaname
```

**Comment by vingerha** [comment]
indeed I could have mentioned this earlier… it is ‘Meteo France’ and you can best use Paris as city.
the sensor works with many weather cards also generic ones so I was also a bit surprised it did not with your card…. I always assumed these were default values for all ‘weather’ sensors

**Comment by Makin-Things** [comment]
Quick update.
Historically the old card which a lot of the logic for this new one is based on used a lot of entities provided by the old dark sky api, so the card really doesn't look at weather.xxxxx for anything.
I had a chat amongst some of the people that have been testing this and we have agreed that it would be good to allow the use of the weather.xxxxx info. It's a bit of work so it won't happen straight away (maybe I need about a week) but I intend to add some switches to make various parts of the card use that source for the info. It may be incremental as well as I will probably do it section by section and release a preview version for testing.
That said, the weather.xxxxxx doesn't provide a super rich set of info so you may still find it difficult to populate some parts of the card.
I now have a Paris weather card set up on my test system, so I understand what is lacking.

**Comment by vingerha** [comment]
I am assuming that weather.xxx is the de factor default for HA hence I would expect cards to at least (!) work with that. Then I also agree that it is not super rich so one could add additional sensors when these are available, e.g. where I live there is no 'one hour rainforecast' ... 100km to the west there is. So having the option to select sensors per items is nice !
Wrt to timing, I have my own card (fork from bramkarten) and this is fine, I only offered some testing/feedback as you requested it :)

**Comment by jeffjensen** [comment]
I'm in the US and using NWS feed - has the same data format and my HA has similar display issues.  I'm looking forward to the updates as well!  Thank you.

**Comment by spijet** [comment]
Another example of similar "standard" weather integration would be [Met.no](https://www.home-assistant.io/integrations/met/). See screenshot below.

<details>

<p>

![weather_entity](https://user-images.githubusercontent.com/14221126/177471702-8916b82f-4051-4be7-af34-468e4c084074.png)

</p>

</details>

**Comment by erkr** [comment]
Good to see the defection weather sensors will become supported. To help you with some examples. I live in the Netherlands where I also use met.no for the weather. 
That integration does allow me to create both a daily and a hourly sensor. These are the data examples. 
Daily:
``` 
temperature: 17
humidity: 55
pressure: 1024.3
wind_bearing: 324.7
wind_speed: 16.6
forecast:
  - condition: cloudy
    precipitation: 0
    temperature: 22.4
    templow: 10.4
    datetime: '2022-07-06T10:00:00+00:00'
    wind_bearing: 317.3
    wind_speed: 15.8
  - condition: cloudy
    precipitation: 0.7
    temperature: 19.8
    templow: 15.4
    datetime: '2022-07-07T10:00:00+00:00'
    wind_bearing: 320.9
    wind_speed: 24.8
  - condition: cloudy
    precipitation: 0.1
    temperature: 20.1
    templow: 11.4
    datetime: '2022-07-08T10:00:00+00:00'
    wind_bearing: 341.4
    wind_speed: 12.2
  - condition: rainy
    precipitation: 1.2
    temperature: 23.2
    templow: 13.5
    datetime: '2022-07-09T10:00:00+00:00'
    wind_bearing: 290
    wind_speed: 19.8
  - condition: cloudy
    precipitation: 0.3
    temperature: 18.7
    templow: 13.9
    datetime: '2022-07-10T10:00:00+00:00'
    wind_bearing: 344.6
    wind_speed: 15.5
attribution: >-
  Weather forecast from met.no, delivered by the Norwegian Meteorological
  Institute.
friendly_name: Home Daily
``` 
Hourly:
``` 
temperature: 15.3
humidity: 69
pressure: 1026.5
wind_bearing: 312.8
wind_speed: 11.9
forecast:
  - condition: cloudy
    precipitation: 0
    temperature: 16.9
    datetime: '2022-07-06T08:00:00+00:00'
    wind_bearing: 320.8
    wind_speed: 11.5
  - condition: cloudy
    precipitation: 0
    temperature: 18
    datetime: '2022-07-06T09:00:00+00:00'
    wind_bearing: 326.2
    wind_speed: 11.9
  - condition: cloudy
    precipitation: 0
    temperature: 19
    datetime: '2022-07-06T10:00:00+00:00'
    wind_bearing: 326.6
    wind_speed: 12.6
  - condition: cloudy
    precipitation: 0
    temperature: 20.5
    datetime: '2022-07-06T11:00:00+00:00'
    wind_bearing: 323.9
    wind_speed: 13.3
  - condition: cloudy
    precipitation: 0
    temperature: 21
    datetime: '2022-07-06T12:00:00+00:00'
    wind_bearing: 318.4
    wind_speed: 12.6
  - condition: cloudy
    precipitation: 0
    temperature: 21.3
    datetime: '2022-07-06T13:00:00+00:00'
    wind_bearing: 309.7
    wind_speed: 10.4
  - condition: cloudy
    precipitation: 0
    temperature: 22.4
    datetime: '2022-07-06T14:00:00+00:00'
    wind_bearing: 304.8
    wind_speed: 12.2
  - condition: cloudy
    precipitation: 0
    temperature: 22
    datetime: '2022-07-06T15:00:00+00:00'
    wind_bearing: 294.3
    wind_speed: 14.4
  - condition: cloudy
    precipitation: 0
    temperature: 21.8
    datetime: '2022-07-06T16:00:00+00:00'
    wind_bearing: 291.5
    wind_speed: 15.5
  - condition: cloudy
    precipitation: 0
    temperature: 21.2
    datetime: '2022-07-06T17:00:00+00:00'
    wind_bearing: 288
    wind_speed: 14
  - condition: cloudy
    precipitation: 0
    temperature: 20.9
    datetime: '2022-07-06T18:00:00+00:00'
    wind_bearing: 287.2
    wind_speed: 13.7
  - condition: partlycloudy
    precipitation: 0
    temperature: 19.5
    datetime: '2022-07-06T19:00:00+00:00'
    wind_bearing: 288.3
    wind_speed: 9.4
  - condition: cloudy
    precipitation: 0
    temperature: 17.5
    datetime: '2022-07-06T20:00:00+00:00'
    wind_bearing: 250.5
    wind_speed: 7.9
  - condition: partlycloudy
    precipitation: 0
    temperature: 17.2
    datetime: '2022-07-06T21:00:00+00:00'
    wind_bearing: 248.6
    wind_speed: 9.7
  - condition: partlycloudy
    precipitation: 0
    temperature: 16.3
    datetime: '2022-07-06T22:00:00+00:00'
    wind_bearing: 245.7
    wind_speed: 10.1
  - condition: sunny
    precipitation: 0
    temperature: 16
    datetime: '2022-07-06T23:00:00+00:00'
    wind_bearing: 241.3
    wind_speed: 10.8
  - condition: cloudy
    precipitation: 0
    temperature: 16.1
    datetime: '2022-07-07T00:00:00+00:00'
    wind_bearing: 236.2
    wind_speed: 12.2
  - condition: cloudy
    precipitation: 0
    temperature: 16.9
    datetime: '2022-07-07T01:00:00+00:00'
    wind_bearing: 246.7
    wind_speed: 15.1
  - condition: partlycloudy
    precipitation: 0
    temperature: 16.8
    datetime: '2022-07-07T02:00:00+00:00'
    wind_bearing: 266.4
    wind_speed: 14
  - condition: rainy
    precipitation: 0.1
    temperature: 16.7
    datetime: '2022-07-07T03:00:00+00:00'
    wind_bearing: 281.4
    wind_speed: 12.6
  - condition: rainy
    precipitation: 0.5
    temperature: 16.7
    datetime: '2022-07-07T04:00:00+00:00'
    wind_bearing: 286.9
    wind_speed: 15.8
  - condition: cloudy
    precipitation: 0
    temperature: 16.2
    datetime: '2022-07-07T05:00:00+00:00'
    wind_bearing: 297
    wind_speed: 19.4
  - condition: rainy
    precipitation: 0.2
    temperature: 16.2
    datetime: '2022-07-07T06:00:00+00:00'
    wind_bearing: 300.2
    wind_speed: 18.4
  - condition: rainy
    precipitation: 0.1
    temperature: 16.1
    datetime: '2022-07-07T07:00:00+00:00'
    wind_bearing: 309.1
    wind_speed: 20.2
attribution: >-
  Weather forecast from met.no, delivered by the Norwegian Meteorological
  Institute.
friendly_name: Home Hourly
```

**Comment by Makin-Things** [comment]
You should be able to set the entities used on the daily forecast section of the card to fix the bottom part.
![image](https://user-images.githubusercontent.com/91187456/177522454-51d69f26-5cc6-4d4c-a807-08613da49d0a.png)

**Comment by Makin-Things** [comment]
@erkr does met.no really forecast wind conditions and if so with this accuracy?
![image](https://user-images.githubusercontent.com/91187456/177522991-740a2374-86e7-47b0-985b-5d4e7ccb1c03.png)
That seems ludicrous and never going to even be close to correct considering wind speed and direction changes constantly.

**Comment by Makin-Things** [comment]
Looking at the data you posted above. You don't get a probability of rain so change slot R1 to ``possible today``, that will get rid of the --- which is where the % chance of rain goes with the defaults. Your integration doesn't provide UV or Fire Danger so change slots R3 and R4 to blank/remove or some other data you have.   You should also be able to get a wind bearing to show up, but with the granularity of the degrees that might be confusing things (I've never seen an integration use more than 16 directions).

**Comment by erkr** [comment]
> @erkr does met.no really forecast wind conditions and if so with this accuracy?
> 

That is what offered as data in the met.no weather sensor.

**Comment by erkr** [comment]
Thanks for your feedback. Try to get a little further on this.
A few remarks: 
- Why do I get NaN for the temperature at the top right. It's present in the met.no weather entity attributes.
- Why is the feeling temperature a required entity?
- some domain and / or class filtering on the entity selections works be helpful

Succes Eric

**Comment by erkr** [comment]
> You should be able to set the entities used on the daily forecast section of the card to fix the bottom part.

That doesn't work! When I add two lines to the config, the card disappears:
``` 
entity_forecast_high_temp_1: weather.home57
entity_forecast_low_temp_1: weather.home57
```

**Comment by Makin-Things** [comment]
The NaN thing is a bug. I missed something in all the changes.
Feeling like is not required it is just a label, it is being changed to optional.
I have done as much domain filtering as possible and trying to limit on classes within a domain will result in people not being able to select the correct sensors.

Try 0.0.3b1.
I set up met.no to test using Amsterdam as the location.
This is what I get.
![image](https://user-images.githubusercontent.com/91187456/177539022-48846cc3-8209-4568-b51b-e7d4d79b2ad6.png)

Using this config
```
card_config_version: 4
daily_extended_forecast_days: 7
daily_forecast_days: 5
daily_forecast_layout: horizontal
entity_current_conditions: weather.amsterdam
entity_forecast_high_temp_1: weather.amsterdam
entity_forecast_icon_1: weather.amsterdam
entity_forecast_low_temp_1: weather.amsterdam
entity_forecast_max: weather.amsterdam
entity_forecast_min: weather.amsterdam
entity_humidity: weather.amsterdam
entity_pop: weather.amsterdam
entity_pos_1: weather.amsterdam
entity_possible_today: weather.amsterdam
entity_possible_tomorrow: weather.amsterdam
entity_pressure: weather.amsterdam
entity_sun: sun.sun
entity_temperature: weather.amsterdam
entity_visibility: weather.amsterdam
entity_wind_bearing: weather.amsterdam
entity_wind_speed: weather.amsterdam
entity_wind_speed_kt: weather.amsterdam
extended_use_attr: false
option_daily_show_extended: true
option_icon_set: hybrid
option_static_icons: false
option_tooltips: false
section_order:
  - title
  - overview
  - extended
  - slots
  - daily_forecast
show_section_extended: false
show_section_title: true
slot_l1: forecast_max
slot_l2: forecast_min
slot_l3: wind
slot_l4: pressure
slot_l5: sun_next
slot_l6: remove
slot_l7: remove
slot_l8: remove
slot_r1: possible_today
slot_r2: possible_tomorrow
slot_r3: humidity
slot_r4: empty
slot_r5: sun_following
slot_r6: remove
slot_r7: remove
slot_r8: remove
text_card_title: Amsterdam
type: custom:platinum-weather-card
```

**Comment by spijet** [comment]
Can confirm that `0.0.3b1` works, albeit with some hiccups (e.g. current condition text is shown "as seen in weather integration's YAML").

Here's how it looks with Met.no data for my city (HA's default weather card shown below for comparison):
<details>
<p>
<img width="401" alt="Screen Shot 2022-07-06 at 21 01 15" src="https://user-images.githubusercontent.com/14221126/177556219-b3117802-44d9-4b28-859d-59bcfb3242e5.png">
</p>
</details>

And here's the card configuration:
<details>
<p>

```yaml
card_config_version: 4
entity_current_conditions: weather.home
entity_current_text: weather.home
entity_forecast_high_temp_1: weather.home
entity_forecast_icon_1: weather.home
entity_forecast_low_temp_1: weather.home
entity_forecast_max: weather.home
entity_forecast_min: weather.home
entity_humidity: weather.home
entity_pop: weather.home
entity_pos_1: weather.home
entity_possible_today: weather.home
entity_pressure: weather.home
entity_temperature: weather.home
entity_wind_bearing: weather.home
entity_wind_gust: sensor.hacs
entity_wind_speed: weather.home
option_icon_set: new
option_static_icons: false
option_time_format: 24hour
section_order:
  - title
  - overview
  - extended
  - slots
  - daily_forecast
slot_l1: forecast_min
slot_l2: wind
slot_l3: remove
slot_l4: remove
slot_l5: remove
slot_l6: remove
slot_l7: remove
slot_l8: remove
slot_r1: forecast_max
slot_r2: humidity
slot_r3: remove
slot_r4: remove
slot_r5: remove
slot_r6: remove
slot_r7: remove
slot_r8: remove
type: custom:platinum-weather-card
```

</p>
</details>

**Comment by Makin-Things** [comment]
@spijet there is nothing that can be done about that. If the integration doesn't provide an user friendly version of the text I can't create it.
tbh. the way that HA forces mapping of real conditions provided by the various bureau's to what it expects simply because of what icons are available in mdi icons is rather poor and the defined format for the weather entities doesn't encourage any form of decent text to be included. Some integrations work around this by creating different entities even though they could 'technically' be included in an attribute of the weather entity. But the bottom line is that I have no control over the data they provide.

**Comment by Makin-Things** [comment]
>     * for some entities, selecting the weather sensor didn't work (managed them via code editing, but now I can't recall which of them failed)
If you aren't able to select weather.xxxx for an entity it is because there is no data for that field in the standard weather integrations. What is in that entity is quite limited and most of it is marked as *optional* in the documentation from HA. Changing it in yaml isn't going to help and will probably have weird consequences.
>     * for a few entities, there is an option to select attributes. This could be a generic option. I.e for the last update time text, an entity needs to be configured. But last update is an attribute for all HA sensors.
>       The code syntax for all entities could be: `entity|attribute` where `|attribute` is optional.
Not going to happen as the HA entity/attribute pickers do not support this. For most entities the information you want is in the entities state.
>     * the daily forecast of met.no includes the wind information. But how to add them to the daily forecast? Don't see an entity option for wind.
Wind in forecasts is not supported and to be honest wind forecasts as a single direction and speed for a 24 hour period makes no sense.
>     * visually removing an entity fails. Need to do that via code:
This is not reproducible. Works perfectly for me.
>     * the sensor names are really not clear: high/low (temp, pressure...), pop, pos, pop_1
This will be documented at some point, but you shouldn't need to ever look at the yaml so what they are called should not be important.
>     * although the weather sensor contains a 5 day forecast, only 4 are displayed
The first of these is today, so there are only 4 future days.

**Comment by vingerha** [comment]
Working much better indeed fro MeteoFrance
not working: daily precipitation, used both main and subsensor
Unsure: daily section starts tomorrow, is this by choice?

**Comment by erkr** [comment]
Hi @Makin-Things 
Thanks for your elaborate answer.
A few remark:
The forecast is 5 days not four. As the header is the info of today. The first forecast entry  is 2022-07-08, =Friday:
``` 

temperature: 17.8
humidity: 71
pressure: 1021.9
wind_bearing: 318.7
wind_speed: 25.9
forecast:
  - condition: cloudy
    precipitation: 0
    temperature: 21.7
    templow: 11.7
    datetime: '2022-07-08T10:00:00+00:00'
    wind_bearing: 340.8
    wind_speed: 13
  - condition: rainy
    precipitation: 0.8
    temperature: 22.3
    templow: 11.6
    datetime: '2022-07-09T10:00:00+00:00'
    wind_bearing: 309.8
    wind_speed: 21.6
  - condition: cloudy
    precipitation: 0
    temperature: 20.6
    templow: 13.6
    datetime: '2022-07-10T10:00:00+00:00'
    wind_bearing: 346.6
    wind_speed: 13.7
  - condition: partlycloudy
    precipitation: 0.1
    temperature: 23.7
    templow: 13.6
    datetime: '2022-07-11T10:00:00+00:00'
    wind_bearing: 347.1
    wind_speed: 8.6
  - condition: partlycloudy
    precipitation: 0
    temperature: 28.3
    templow: 15.3
    datetime: '2022-07-12T10:00:00+00:00'
    wind_bearing: 191.6
    wind_speed: 7.9
attribution: >-
  Weather forecast from met.no, delivered by the Norwegian Meteorological
  Institute.
``` 
As you can see, the last entry, 2022-07-12 =Tuesday, is missing:
![image](https://user-images.githubusercontent.com/4363599/177739524-4c81bfe4-dfe9-47c2-86fd-bda425ff9505.jpeg)

**Comment by Makin-Things** [comment]
@vingerha @erkr  The daily forecast section starts from tomorrow as the details for today (possibly in better detail, depending on what attributes your weather integration provides or you can create as entities from other data sources). Duplicating that information seems rather pointless. Referring to the image immediately above, if there was data for Tues (the 5th day) it would be shown. The card is just being a little intelligent and when it gets to a day where there is no data it aborts and formats the data it has to use the space. I have seen some integrations that have a varying number of days dependant on the time of day, so this seems like the best way to deal with this. From looking at the data the met.no integration provides I would be looking around to see if there was a better provider.
I have thought about starting the daily forecasts from today, but I think that should probably only occur if the slots section is disabled. Interested in opinions.

In Australia we are lucky to have an awesome weather integration written by someone else, but I now look after it and do enhancements and deal with issues. It would be really nice if other countries has similar integrations.

Also I released 0.0.3b2 earlier to try and improve the formatting of the short text and fix something I missed in the tooltips for the horizontal forecast.

**Comment by Makin-Things** [comment]
I have released version 0.0.3 which I think addresses the majority of the issues and comments in this issue.
I am closing it now. Please create a new issue of there is still anything that is not working.

**Comment by erkr** [comment]
@Makin-Things 
There is really one day missing and the data is one day wrong. The data for Wednesday is on Tuesday in your card:
![image](https://user-images.githubusercontent.com/4363599/177937218-f077cfc5-79fc-4d45-9619-fd97aeb1b5c0.jpeg)
